server/chat-commands.js

```diff
@@ -593,13 +593,14 @@ const commands = {
 	blockpm: 'ignorepms',
 	blockpms: 'ignorepms',
 	ignorepm: 'ignorepms',
-	ignorepms(target, room, user) {
+	ignorepms(target, room, user, connection) {
 		if (user.ignorePMs === (target || true)) return this.errorReply("You are already blocking private messages! To unblock, use /unblockpms");
 		user.ignorePMs = true;
 		if (target in Config.groups) {
 			user.ignorePMs = target;
 			return this.sendReply(`You are now blocking private messages, except from staff and ${target}.`);
 		}
+		connection.send(user.getUpdateuserText());
```

Comment:
Why not `user.send`?

Heck, why not `user.update()`?

Reply:
The purpose of this is to get all the user's connections in sync, right?

Reply:
> The purpose of this is to get all the user's connections in sync, right?

 Yes, sorry, I dont know much (anything?) about non-sim code. What's the preferred way to do this?

Reply:
`connection.send` will only send a message to the connection that sent the connection.

`user.send` will send a message to all connections associated with a user. Which is the one you want here, because you want all connections (and therefore all clients) to know that the user's settings have been updated.

I also think `user.send(user.getUpdateuserText());` should be broken out into its own function `user.update();`